src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.cs

```diff
@@ -124,7 +124,14 @@ protected virtual void OnToggle()
 
 		internal void AutomationPeerToggle()
 		{
+			var oldValue = IsChecked;
 			OnToggle();
+			var newValue = IsChecked;
+
+			if ((!oldValue.HasValue || !oldValue.Value) && newValue == true)
+			{
+				InvokeCommand();
```

Comment:
I'm not too familiar with the AutomationPeer mechanics, but is there any risk of the command being called twice? @ajpinedam I think you mentioned that without the fix, the command was still being raised for a particular sub-area of the touch target of the toggle button - would it get raised once or twice in that case?

Reply:
@davidjohnoliver Tested it and indeed we could see cases when the Command is executed twice. We will be merging this code as it and created a follow-up issue https://github.com/unoplatform/uno/issues/6535 . 

Also, @MartinZikmund  is working on this control and might be fixing this issue